libs/balances-and-allowances/src/hooks/useAddPriorityAllowance.ts

```diff
@@ -0,0 +1,47 @@
+import { useSetAtom } from 'jotai/index'
+import { useCallback } from 'react'
+
+import { Erc20, ERC_20_INTERFACE } from '@cowprotocol/abis'
+import { GP_VAULT_RELAYER } from '@cowprotocol/common-const'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { Contract } from '@ethersproject/contracts'
+import { useWeb3React } from '@web3-react/core'
+
+import { allowancesFullState } from '../state/allowancesAtom'
+
+interface PriorityAllowanceParams {
+  account: string | undefined
+  chainId: SupportedChainId
+  tokenAddress: string
+  blockNumber: number
+}
+
+export function useAddPriorityAllowance() {
+  const { provider } = useWeb3React()
+  const setAllowance = useSetAtom(allowancesFullState)
+
+  return useCallback(
+    ({ chainId, account, tokenAddress, blockNumber }: PriorityAllowanceParams) => {
+      if (!provider || !account) return undefined
+
+      const spender = GP_VAULT_RELAYER[chainId]
+      const tokenContract = new Contract(tokenAddress, ERC_20_INTERFACE, provider) as Erc20
+
+      if (!blockNumber) return
```

Comment:
Nitpick: any reason for not returning when no `blockNumber` on line `25`?

Reply:
It's just a protection from invalid RPC-node response.
Just in case if the transaction receipt doesn't contain `blockNumber` (what should not happen in most cases)

Reply:
I'm not arguing against the check.
Just suggesting to put it together with `!provider || !account`

```suggestion
      if (!provider || !account || !blockNumber) return undefined

      const spender = GP_VAULT_RELAYER[chainId]
      const tokenContract = new Contract(tokenAddress, ERC_20_INTERFACE, provider) as Erc20
```

Reply:
Thanks! Fixed